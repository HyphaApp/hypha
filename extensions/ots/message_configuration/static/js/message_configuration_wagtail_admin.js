"use strict";

(function ($) {
    function add_update_events() {
        $(".message_type").each(function () {
            var message_type = $(this);
            var textarea = message_type
                .siblings(".email_message")
                .find("textarea");
            var select = message_type.find("select");
            if (!select.attr("evented")) {
                select.attr("evented", true);
                select.change(function () {
                    var message_id = $(this).val();
                    var message = $(
                        ".messaging_help_panel [attr-email-id='" +
                            message_id +
                            "']"
                    )
                        .text()
                        .trim();
                    textarea.val(message);
                });
            }
        });
    }

    $(function () {
        $("#id_messaging_settings-ADD").click(function () {
            add_update_events();
        });
        add_update_events();
    });
})(jQuery);
