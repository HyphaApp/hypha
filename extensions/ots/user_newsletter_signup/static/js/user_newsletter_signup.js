"use strict";

(function ($) {
    $(function () {
        $(".profile_newsletter_signup").click(function () {
            window.alert("You clicked me!");
        });
    });
})(jQuery);
