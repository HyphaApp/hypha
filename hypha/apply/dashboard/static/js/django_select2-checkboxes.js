(function ($) {
    $.fn.select2.amd.require(
        [
            "select2/multi-checkboxes/selection",
            "select2/multi-checkboxes/results",
        ],
        function (SelectionAdapter, ResultsAdapter) {
            $(function () {
                $(".django-select2-checkboxes").each(function (i, element) {
                    var $element = $(element);
                    $element.select2({
                        placeholder: $element.data("placeholder"),
                        closeOnSelect: false,
                        templateSelection: function (data) {
                            let filterType = $element.data("placeholder");

                            if (!data.selected.length) {
                                return filterType;
                            } else if (
                                data.selected.length == data.all.length
                            ) {
                                return "All " + filterType + " selected";
                            }
                            return (
                                data.selected.length +
                                " of " +
                                data.all.length +
                                " " +
                                filterType
                            );
                        },
                        selectionAdapter: SelectionAdapter,
                        returnesultsAdapter: ResultsAdapter,
                    });
                });
            });
        }
    );
})(this.jQuery);
