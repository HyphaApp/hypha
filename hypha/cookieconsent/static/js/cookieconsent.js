(function () {
    "use strict";

    if (typeof Cookies !== "undefined") {
        const cookieconsent = document.querySelector(".cookieconsent");

        if (
            typeof Cookies.get("cookieconsent") === "undefined" &&
            cookieconsent
        ) {
            cookieconsent.classList.add("js-cookieconsent-open");
        }

        const cookie_buttons = Array.prototype.slice.call(
            document.querySelectorAll("button[data-consent]")
        );
        const sitedomain = window.location.hostname.split(".").slice(-2);
        const cookiedomain = sitedomain.join(".");
        let cookie_options = [];
        cookie_options["domain"] = cookiedomain;
        cookie_options["sameSite"] = "strict";
        cookie_options["expires"] = 365;
        if (window.location.protocol === "https:") {
            cookie_options["secure"] = true;
        }

        cookie_buttons.forEach(function (button) {
            button.addEventListener("click", function () {
                if (button.getAttribute("data-consent") == "true") {
                    Cookies.set("cookieconsent", "accept", cookie_options);
                } else {
                    Cookies.set("cookieconsent", "decline", cookie_options);
                }
                cookieconsent.classList.remove("js-cookieconsent-open");
            });
        });
    }
})();
