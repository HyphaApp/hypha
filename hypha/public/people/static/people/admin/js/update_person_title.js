$(document).ready(function () {
    var $lastNameInput = $("#id_last_name");
    var $firstNameInput = $("#id_first_name");
    var $titleInput = $("#id_title");
    var $slugInput = $("#id_slug");

    $firstNameInput.on("input", function () {
        joinFirstNameLastName();
    });

    $lastNameInput.on("input", function () {
        joinFirstNameLastName();
    });

    function joinFirstNameLastName() {
        var firstName = $firstNameInput.val();
        var lastName = $lastNameInput.val();
        var title = firstName + " " + lastName;

        $slugInput.data("previous-val", $slugInput.val());
        $titleInput.data("previous-val", $titleInput.val());
        $titleInput.val(title);
        $titleInput.blur(); // Trigger slug update
    }
});
