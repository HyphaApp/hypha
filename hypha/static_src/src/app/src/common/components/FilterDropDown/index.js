import React from 'react';
import PropTypes from 'prop-types';
// import "./styles.scss";
import Select from '@material-ui/core/Select';
import InputLabel from '@material-ui/core/InputLabel';
import MenuItem from '@material-ui/core/MenuItem';
import FormControl from '@material-ui/core/FormControl';
import { withStyles } from '@material-ui/core/styles';
import Checkbox from '@material-ui/core/Checkbox';
import ListItemText from '@material-ui/core/ListItemText';
import Input from '@material-ui/core/Input';

const styles = {
  formControl: {
    width: '100%',
    marginRight: 10,
    height: 40,
  }
}

class FilterDropDown extends React.PureComponent {

  render() {
    const { filter, value, handleChange, renderValues, classes } = this.props;
    return  <FormControl 
              variant="outlined" 
              key={filter.label} 
              size={"small"}
              classes={{  root : classes.formControl }}
              >
                <InputLabel>{filter.label}</InputLabel>
                <Select
                  multiple
                  name={filter.filterKey}
                  value={value} 
                  onChange={handleChange}
                  input={<Input />}
                  renderValue={(selected) => renderValues(selected, filter)}
                >
                  {filter["options"].map(
                    option => <MenuItem 
                      value={option.key} 
                      key={option.key}
                      >
                      <Checkbox 
                        checked={value.indexOf(option.key) > -1}
                        style ={{ color: "#0c72a0b3" }}
                      />
                      <ListItemText primary={option.label} />
                    </MenuItem>
                    )}
                </Select>
              </FormControl>
          }
}


FilterDropDown.propTypes = {
  filter: PropTypes.object,
  value: PropTypes.array,
  handleChange: PropTypes.func,
  renderValues: PropTypes.func,
  classes: PropTypes.object
}

export default withStyles(styles)(FilterDropDown);
