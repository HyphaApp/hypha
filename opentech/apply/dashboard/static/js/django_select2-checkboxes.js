(function ($) {
    var init = function ($element) {
        options = {
            placeholder: $element.data('placeholder'),
            templateSelection: function(selected, total) {
                let filterType = this.placeholder;
                if ( !selected.length ) {
                    return filterType;
                } else if ( selected.length===total ) {
                    return 'All ' + filterType + ' selected';
                }
                return selected.length + ' of ' + total + ' ' + filterType + ' selected';
            }
        };
        $element.select2MultiCheckboxes(options);
    };
    $(function () {
        $('.django-select2-checkboxes').each(function (i, element) {
            var $element = $(element);
            init($element);
        });
    });
}(this.jQuery));
