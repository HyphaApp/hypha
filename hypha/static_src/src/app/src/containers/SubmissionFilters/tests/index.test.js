import React from "react";
import {mount} from "enzyme";
import * as enzyme from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import  {SubmissionFiltersContainer}  from '../index.js';
import LoadingPanel from '@components/LoadingPanel';
import FilterDropDown from '@common/components/FilterDropDown'
import * as Immutable from 'seamless-immutable';
enzyme.configure({adapter: new Adapter()});

describe("Test submission filter Container", () => {
    it("Should render submission filter with loading true", () => {
        const initializeAction = jest.fn()
        const wrapper = mount(
                <SubmissionFiltersContainer submissionFilters={{loading : true}} initializeAction={initializeAction}/>
        );
        expect(wrapper.find('.filter-container').length).toEqual(0)
        expect(initializeAction).toHaveBeenCalled()
        expect(wrapper.containsMatchingElement(<LoadingPanel />)).toBe(true)
        expect(wrapper).toMatchSnapshot();
    });

    it("Should render submission filter with loading false", () => {
        const initializeAction = jest.fn()
        const updateSelectedFilter = jest.fn()
        const updateFilterQuery = jest.fn()
        const onFilter = jest.fn()
        const deleteSelectedFilters = jest.fn()

        const doNotRender = []
        const submissionFilters = Immutable.from({
            loading : false,
            selectedFilters : {
                1 : []
            },
            filters :[{filterKey : "key1", label : "label1", options: [{key : 1, label: "label2"}]}]
        })
        const classes = {
            filterButton : "string"
        }
        const wrapper = shallow(
                <SubmissionFiltersContainer 
                submissionFilters={submissionFilters} 
                initializeAction={initializeAction}
                updateSelectedFilter={updateSelectedFilter}
                updateFilterQuery={updateFilterQuery}
                onFilter={onFilter}
                doNotRender={doNotRender}
                deleteSelectedFilters={deleteSelectedFilters}
                classes={classes}
                getFiltersToBeRendered={[{filterKey : "key1", label : "label1", options: [{key : 1, label: "label2"}]}]}
                />
        );

        expect(wrapper.find('.filter-container').length).toEqual(1)
        expect(initializeAction).toHaveBeenCalled()
        expect(wrapper.containsMatchingElement(<LoadingPanel />)).toBe(false)
        expect(wrapper.containsMatchingElement(<FilterDropDown />)).toBe(true)
        const subObj = wrapper.instance()
        subObj.handleChange({target : { name : "name1", value : "value1" } })
        expect(updateSelectedFilter).toHaveBeenCalled()
        expect(updateSelectedFilter).toHaveBeenCalledWith("name1", "value1")
        expect(wrapper).toMatchSnapshot();
    });
});
