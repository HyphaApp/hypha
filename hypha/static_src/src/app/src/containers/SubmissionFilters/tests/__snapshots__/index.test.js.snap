// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test submission filter Container Should render submission filter with loading false 1`] = `
<div
  className="filter-container"
>
  <WithStyles(FilterDropDown)
    filter={
      Object {
        "filterKey": "key1",
        "label": "label1",
        "options": Array [
          Object {
            "key": 1,
            "label": "label2",
          },
        ],
      }
    }
    handleChange={[Function]}
    key="label1"
    renderValues={[Function]}
    value={Array []}
  />
  <WithStyles(ForwardRef(Button))
    classes={
      Object {
        "root": "string",
      }
    }
    onClick={[Function]}
    size="small"
    variant="contained"
  >
    Filter
  </WithStyles(ForwardRef(Button))>
  <WithStyles(ForwardRef(Snackbar))
    TransitionComponent={[Function]}
    anchorOrigin={
      Object {
        "horizontal": "center",
        "vertical": "bottom",
      }
    }
    autoHideDuration={6000}
    key="bottomcenter"
    onClose={[Function]}
    open={false}
  >
    <AlertComponent
      onClose={[Function]}
      severity="success"
    >
      URL copied to Clipboard!
    </AlertComponent>
  </WithStyles(ForwardRef(Snackbar))>
  <WithStyles(ForwardRef(Tooltip))
    placement="bottom"
    title={
      <span
        style={
          Object {
            "fontSize": "15px",
          }
        }
      >
        clear
      </span>
    }
  >
    <Memo(ForwardRef(HighlightOffIcon))
      className="delete-button"
      fontSize="large"
      onClick={[Function]}
      style={
        Object {
          "visibility": "visible",
        }
      }
    />
  </WithStyles(ForwardRef(Tooltip))>
</div>
`;

exports[`Test submission filter Container Should render submission filter with loading true 1`] = `
<SubmissionFiltersContainer
  initializeAction={
    [MockFunction] {
      "calls": Array [
        Array [],
      ],
    }
  }
  submissionFilters={
    Object {
      "loading": true,
    }
  }
>
  <LoadingPanel>
    <div
      className="loading-panel"
    >
      <div
        className="loading-panel__text"
      >
        <h5>
          Loading...
        </h5>
      </div>
      <div
        className="loading-panel__icon"
      >
        <OTFLoadingIcon>
          <svg
            className="logo-loading"
            viewBox="0 0 45 40"
          >
            <g
              fill="none"
              fillRule="nonzero"
            >
              <rect
                className="logo-part logo-part-1"
                fill="#25AAE1"
                height="5"
                transform="translate(10 0)"
                width="5"
              />
              <rect
                className="logo-part logo-part-2"
                fill="#25AAE1"
                height="5"
                transform="translate(5 5)"
                width="5"
              />
              <rect
                className="logo-part logo-part-3"
                fill="#25AAE1"
                height="5"
                transform="translate(0 10)"
                width="5"
              />
              <rect
                className="logo-part logo-part-4"
                fill="#25AAE1"
                height="5"
                transform="translate(0 15)"
                width="5"
              />
              <rect
                className="logo-part logo-part-5"
                fill="#25AAE1"
                height="5"
                transform="translate(0 20)"
                width="5"
              />
              <rect
                className="logo-part logo-part-6"
                fill="#25AAE1"
                height="5"
                transform="translate(0 25)"
                width="5"
              />
              <rect
                className="logo-part logo-part-7"
                fill="#25AAE1"
                height="5"
                transform="translate(5 30)"
                width="5"
              />
              <rect
                className="logo-part logo-part-8"
                fill="#25AAE1"
                height="5"
                transform="translate(10 35)"
                width="5"
              />
              <rect
                className="logo-part logo-part-9"
                fill="#25AAE1"
                height="5"
                transform="translate(30 0)"
                width="5"
              />
              <rect
                className="logo-part logo-part-10"
                fill="#25AAE1"
                height="5"
                transform="translate(35 5)"
                width="5"
              />
              <rect
                className="logo-part logo-part-11"
                fill="#25AAE1"
                height="5"
                transform="translate(40 10)"
                width="5"
              />
              <rect
                className="logo-part logo-part-12"
                fill="#25AAE1"
                height="5"
                transform="translate(40 15)"
                width="5"
              />
              <rect
                className="logo-part logo-part-13"
                fill="#25AAE1"
                height="5"
                transform="translate(40 20)"
                width="5"
              />
              <rect
                className="logo-part logo-part-14"
                fill="#25AAE1"
                height="5"
                transform="translate(40 25)"
                width="5"
              />
              <rect
                className="logo-part logo-part-15"
                fill="#25AAE1"
                height="5"
                transform="translate(35 30)"
                width="5"
              />
              <rect
                className="logo-part logo-part-16"
                fill="#25AAE1"
                height="5"
                transform="translate(30 35)"
                width="5"
              />
            </g>
          </svg>
        </OTFLoadingIcon>
      </div>
    </div>
  </LoadingPanel>
</SubmissionFiltersContainer>
`;
